skylark/skylark.bzl

```diff
@@ -170,113 +171,4 @@ Example:
-)
-"""
-
-SIX_BUILD_FILE = """
-py_library(
-    name = "six",
-    srcs = ["six.py"],
-    srcs_version = "PY2AND3",
-    visibility = ["//visibility:public"],
-)
-"""
-
-def skydoc_repositories():
-    """Adds the external repositories used by the skylark rules."""
-    git_repository(
-        name = "protobuf",
-        remote = "https://github.com/protocolbuffers/protobuf.git",
-        # Latest tagged version at time of writing is v3.6.1, which doesn't
-        # include fixes for --incompatible_package_name_is_a_function,
-        # --incompatible_new_actions_api, and possibly others.
-        # TODO: Update to a newer tagged version when available.
-        commit = "7b28271a61a3da0a37f6fda399b0c4c86464e5b3",  # 2018-11-16
-    )
-
-    # Protobuf expects an //external:python_headers label which would contain the
-    # Python headers if fast Python protos is enabled. Since we are not using fast
-    # Python protos, bind python_headers to a dummy target.
-    native.bind(
-        name = "python_headers",
-        actual = "//:dummy",
-    )
-
-    http_archive(
-        name = "markupsafe_archive",
-        urls = ["https://pypi.python.org/packages/source/M/MarkupSafe/MarkupSafe-0.23.tar.gz#md5=f5ab3deee4c37cd6a922fb81e730da6e"],
-        sha256 = "a4ec1aff59b95a14b45eb2e23761a0179e98319da5a7eb76b56ea8cdc7b871c3",
-        build_file_content = MARKUPSAFE_BUILD_FILE,
-        strip_prefix = "MarkupSafe-0.23",
-    )
-
-    native.bind(
-        name = "markupsafe",
-        actual = "@markupsafe_archive//:markupsafe",
-    )
-
-    http_archive(
-        name = "jinja2_archive",
-        urls = ["https://pypi.python.org/packages/source/J/Jinja2/Jinja2-2.8.tar.gz#md5=edb51693fe22c53cee5403775c71a99e"],
-        sha256 = "bc1ff2ff88dbfacefde4ddde471d1417d3b304e8df103a7a9437d47269201bf4",
-        build_file_content = JINJA2_BUILD_FILE,
-        strip_prefix = "Jinja2-2.8",
-    )
-
-    native.bind(
-        name = "jinja2",
-        actual = "@jinja2_archive//:jinja2",
-    )
-
-    http_archive(
-        name = "mistune_archive",
-        urls = ["https://pypi.python.org/packages/source/m/mistune/mistune-0.7.1.tar.gz#md5=057bc28bf629d6a1283d680a34ed9d0f"],
-        sha256 = "6076dedf768348927d991f4371e5a799c6a0158b16091df08ee85ee231d929a7",
-        build_file_content = MISTUNE_BUILD_FILE,
-        strip_prefix = "mistune-0.7.1",
-    )
-
-    native.bind(
-        name = "mistune",
-        actual = "@mistune_archive//:mistune",
-    )
-
-    http_archive(
-        name = "six_archive",
-        urls = ["https://pypi.python.org/packages/source/s/six/six-1.10.0.tar.gz#md5=34eed507548117b2ab523ab14b2f8b55"],
-        sha256 = "105f8d68616f8248e24bf0e9372ef04d3cc10104f1980f54d57b2ce73a5ad56a",
-        build_file_content = SIX_BUILD_FILE,
-        strip_prefix = "six-1.10.0",
-    )
-
-    native.bind(
-        name = "six",
-        actual = "@six_archive//:six",
-    )
+skydoc_repositories = _skydoc_repositories
```

Comment:
Why import it as _skydoc_repositories if you are renaming it its default name?

Reply:
Because
1. Loading non-explicitly-exported symbols is disallowed (Without redefining skydoc_repositories in this file, a bzl file depending on this file would not be able to load skydoc_repositories)
2. Overriding an imported global symbol is disallowed
    (I'm not allowed to do skydoc_repositories = skydoc_repositories)

Thus, I have to load it in as a different symbol name then re-export it under the desired name.